vnext/Microsoft.ReactNative/XamlView.cpp

```diff
@@ -25,12 +27,38 @@ comp::Compositor GetCompositor(const XamlView &view) {
     return window.Compositor();
   }
 #ifdef USE_WINUI3
-  else {
-    return TryGetXamlRoot(view).Compositor();
+  else if (auto root = TryGetXamlRoot(view)) {
+    return root.Compositor();
   }
 #endif
+  return GetCompositor();
+}
 
-  throw std::exception("Could not get a compositor instance");
+static DWORD tlsCompositor = 0;
+void SetCompositor(const comp::Compositor &compositor) {
+  if (tlsCompositor == 0) {
```

Comment:
Any reason we can't use the thread_local specifier instead of dealing with Tls functions?

Reply:
good idea!